src/components/topic/Description.js

```diff
@@ -81,21 +114,168 @@ const TeamDescriptiton = styled.div`
+  width: 100%;
+  &&& * {
+    margin-bottom: 0;
+  }
 `
 
+const YoutubeIframeContainer = styled.div`
+  max-width: 100%;
+  margin-bottom: 0;
+  iframe,object,embed {
+    width: 560px;
+    height: 315px;
+  }
+`
+
+const DescTextBlock = styled.div`
+  color: #808080;
+  font-size: 0.88rem;
+  line-height: 1.8;
+  margin: 0.94rem 1.33rem 0 1.33rem;
+}
+`
+
+const TopicDescriptionJSX = {
+  BlockQuote: (content) => {
+    return (
+      <BlockQuote dangerouslySetInnerHTML={{ __html: _.get(content, [ 0 ], '') }}></BlockQuote>
+    )
+  },
+  Youtube: (content) => {
+    let { description, youtubeId } = _.get(content, [ 0 ], {})
+    if (!youtubeId) {
+      return null
+    }
+    return (
+      <YoutubeContainer>
+        <YoutubeIframeContainer>
+          <YoutubePlayer videoId={youtubeId} />
+        </YoutubeIframeContainer>
+        <DescTextBlock>
+          {description}
+        </DescTextBlock>
+      </YoutubeContainer>
+    )
+  }
+}
+
+const TopicTeamJSX = {
+  Image: (content) => {
+    const imgSrc = _.get(content, [ 0, 'tablet', 'url' ], '')
+    return (
+      <img src={imgSrc} />
+    )
+  }
+}
+
+const getTopicContent = (data, contentType) => {
+  if (Array.isArray(data)) {
+    return data.map((ele) => {
+      let styles = {}
+      let type = ele.type
+      let ArticleComponent = null
+      let TopicComponent = null
+      if (contentType === 'description') {
+        switch(type) {
+          case 'blockquote':
+            TopicComponent = TopicDescriptionJSX.BlockQuote
+            break
+          case 'youtube':
+            TopicComponent = TopicDescriptionJSX.Youtube
+            break
+          default:
+            ArticleComponent = getArticleComponent(type)
+        }
+      } else if (contentType === 'team') {
+        switch(type) {
+          case 'image':
+            TopicComponent = TopicTeamJSX.Image
+            break
+          default:
+            ArticleComponent = getArticleComponent(type)
+        }
+      }
+
+      if (!ArticleComponent && !TopicComponent) {
+        return null
+      }
+
+      if (type === 'embeddedcode') {
+        let embeddedContent = _.get(ele, [ 'content', 0 ], {})
+        let width = _.get(embeddedContent, 'width')
+        let height = _.get(embeddedContent, 'height')
+        if (width) {
+          styles.width = width
+        }
+        if (height) {
+          styles.minHeight = height
+        }
+      }
+      return (
+        <StyledTopicComponent
+          key={ele.id}
+          fontSize={fontSize}
+          style={styles}
+          headerOneText={type === 'header-one' ? _.get(ele, 'content.0') : ''}
+          size={contentSize}
+        >
+          {
+            ArticleComponent ?
+              <ArticleComponent
+                alignment={ele.alignment}
+                content={ele.content}
+                id={ele.id}
+                styles={ele.styles}
+              />:TopicComponent(_.get(ele, 'content', []))
```

Comment:
Follow the above comments,
could we rewrite it like
```
<Component
  alignment={ele.alignment}
  content={ele.content}
  id={ele.id}
  styles={ele.styles}
/>
```

Reply:
Sure. Will follow these suggestions in next update.